### PR TITLE
cli -APPLY_TRANS -INVERSE

### PR DESCRIPTION
**Pull request by HeadLessHUN**
it will extend the features of the -APPLY_TRANS command with an optional -INVERSE option.

https://github.com/CloudCompare/CloudCompare/blob/b999cd293309334bec54cbddf3f94e67df360a03/qCC/TODO.txt#L33C6-L33C6

### CODE REVIEW

**Comment by HeadLessHUN** [comment]
Check

**Comment by dgirardeau** [comment]
Have you really 'closed' the PR as in 'I don't want this to be merged anymore'? 😅

File: CHANGELOG.md [hunk 1]
```diff
@@ -248,6 +248,7 @@ v2.13.alpha (???) - (??/??/????)
 		- New sub-option for the -C2M_DIST command: -UNSIGNED, to compute unsigned distances
 		- New sub-option for the -SF_ARITHMETIC command: -IN_PLACE, to update the scalar field in place, without creating a new SF
 		- Most methods using scalar fields as input will now also accept the scalar field name (in lieu of the SF index)
+		- New sub-option for -APPLY_TRANS command: -INVERSE to inverse the transofrmation matrix before apply.
```
Comment (dgirardeau) [inline CHANGELOG.md:251]:
```suggestion
		- New sub-option for -APPLY_TRANS command: -INVERSE to inverse the transformation matrix before it is applied.
```

File: qCC/ccCommandLineCommands.cpp [hunk 2]
```diff
@@ -1909,6 +1928,10 @@ bool CommandApplyTransformation::process(ccCommandLineInterface& cmd)
 	{
 		return cmd.error(QObject::tr("Failed to read transformation matrix file '%1'!").arg(filename));
 	}
+	if (inverse) {
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:1931]:
```suggestion
	if (inverse)
	{
```

File: qCC/ccCommandLineCommands.cpp [hunk 3]
```diff
@@ -1909,6 +1928,10 @@ bool CommandApplyTransformation::process(ccCommandLineInterface& cmd)
...
+		cmd.print(QObject::tr("Transformation before inverse:\n") + mat.toString(6));
+		mat=mat.inverse();
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:1933]:
```suggestion
		mat = mat.inverse();
```

File: qCC/ccCommandLineCommands.cpp [hunk 4]
```diff
@@ -1909,6 +1928,10 @@ bool CommandApplyTransformation::process(ccCommandLineInterface& cmd)
 	{
 		return cmd.error(QObject::tr("Failed to read transformation matrix file '%1'!").arg(filename));
 	}
+	if (inverse) {
+		cmd.print(QObject::tr("Transformation before inverse:\n") + mat.toString(6));
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:1932]:
```suggestion
		cmd.print(QObject::tr("Transformation before inversion:\n") + mat.toString(6));
```

Comment (dgirardeau) [reply qCC/ccCommandLineCommands.cpp:1932]: Did you want to display the inverted matrix as well maybe?

**Review by dgirardeau — CHANGES_REQUESTED** [review]

File: qCC/ccCommandLineCommands.cpp (hunk 4 above)
Comment (HeadLessHUN) [reply qCC/ccCommandLineCommands.cpp:1932]: In normal case the applied matrix shown, so i just needed to add this to show both of them.

**Comment by HeadLessHUN** [comment]
It failed again on the macos build. Can you trigger a recheck without closing/reopening? It's probably some kind of network issue or rate limiter on that site where the plugin source is held.

**Review by dgirardeau — APPROVED** [review]

**Comment by dgirardeau** [comment]
You have to click on the issue, and then you have a 're-run job' option.

**Pull request merged**